File: Baekjoon/투포인터/2003.py
# https://www.acmicpc.net/problem/2003

n, m = map(int, input().split())
a = list(map(int, input().split()))
start, end = 0, 0
cnt = 0

while start <= end and end <= n:
    result = sum(a[start:end])
    if result == m:
        cnt += 1
        start += 1
        end += 1
        continue
    if result <= m:
        end += 1
        continue
    if result > m and start < end:
        start += 1
        continue

print(cnt)

File: Baekjoon/투포인터/2003.cpp
// https://www.acmicpc.net/problem/2003

#include <iostream>
#include <vector>
using namespace std;

int main() {
    int N, M;
    cin >> N >> M;
    vector<int> A(N);
    for (int i = 0; i < N; i++) {
        cin >> A[i];
    }

    int ans = 0;
    for (int i = 0; i < N; i++) {
        int sum = 0;
        for (int j = i; j < N; j++) {
            sum += A[j];
            if (sum == M) {
                ans++;
                break;
            }
        }
    }
    cout << ans << endl;

    return 0;
}